lib/elixir/lib/module.ex

```diff
@@ -1098,13 +1098,15 @@ defmodule Module do
 
   ## Helpers
 
-  defp preprocess_attribute(key, value) when key in [:moduledoc, :typedoc, :doc] do
-    unless match?({line, _} when is_integer(line), value) do
-      raise ArgumentError, "expected #{key} attribute given in the {line, doc} format, got: #{inspect(value)}"
-    end
+  defp preprocess_attribute(key, {line, doc} = value)
```

Comment:
The `unless` already looked as too much complexity IMO, adding more checks to it is only going to make things worse.